File: sdk/dart/lib/src/affiliate.dart
import 'http_client.dart';

/// Affiliate: affiliate program management
class AffiliateResource {
  final HttpClient httpClient;
  AffiliateResource(this.httpClient);

  /// GET /api/affiliates/summary - Get Affiliate summary
  Future<Map<String, dynamic>> getSummary([Map<String, dynamic>? params]) async {
    return httpClient.get('/api/affiliates/summary', params);
  }

  /// GET /api/affiliates/campaigns - Get Affiliate campaigns
  Future<Map<String, dynamic>> listCampaigns([Map<String, dynamic>? params]) async {
    return httpClient.get('/api/affiliates/campaigns', params);
  }

  /// GET /api/affiliates/commissions - Get Affiliate commissions
  Future<Map<String, dynamic>> listCommissions([Map<String, dynamic>? params]) async {
    return httpClient.get('/api/affiliates/commissions', params);
  }

  /// GET /api/affiliates/payouts - Get Affiliate payouts
  Future<Map<String, dynamic>> listPayouts([Map<String, dynamic>? params]) async {
    return httpClient.get('/api/affiliates/payouts', params);
  }

  /// GET /api/affiliates/vouchers - Get Affiliate vouchers
  Future<Map<String, dynamic>> listVouchers([Map<String, dynamic>? params]) async {
    return httpClient.get('/api/affiliates/vouchers', params);
  }

  /// GET /api/affiliates/commissionplans - Get Affiliate commission plans
  Future<Map<String, dynamic>> listCommissionPlans([Map<String, dynamic>? params]) async {
    return httpClient.get('/api/affiliates/commissionplans', params);
  }

}

File: sdk/rust/src/affiliate.rs
//! Affiliate: affiliate program management

use std::sync::Arc;
use crate::http_client::HttpClient;

pub struct AffiliateResource {
    http: Arc<HttpClient>,
}

impl AffiliateResource {
    pub fn new(http: Arc<HttpClient>) -> Self { Self { http } }

    /// GET /api/affiliates/summary - Get Affiliate summary
    pub async fn get_summary(&self, params: Option<&serde_json::Value>) -> Result<serde_json::Value, crate::http_client::ApiError> {
        self.http.get("/api/affiliates/summary", None).await
    }

    /// GET /api/affiliates/campaigns - Get Affiliate campaigns
    pub async fn list_campaigns(&self, params: Option<&serde_json::Value>) -> Result<serde_json::Value, crate::http_client::ApiError> {
        self.http.get("/api/affiliates/campaigns", None).await
    }

    /// GET /api/affiliates/commissions - Get Affiliate commissions
    pub async fn list_commissions(&self, params: Option<&serde_json::Value>) -> Result<serde_json::Value, crate::http_client::ApiError> {
        self.http.get("/api/affiliates/commissions", None).await
    }

    /// GET /api/affiliates/payouts - Get Affiliate payouts
    pub async fn list_payouts(&self, params: Option<&serde_json::Value>) -> Result<serde_json::Value, crate::http_client::ApiError> {
        self.http.get("/api/affiliates/payouts", None).await
    }

    /// GET /api/affiliates/vouchers - Get Affiliate vouchers
    pub async fn list_vouchers(&self, params: Option<&serde_json::Value>) -> Result<serde_json::Value, crate::http_client::ApiError> {
        self.http.get("/api/affiliates/vouchers", None).await
    }

    /// GET /api/affiliates/commissionplans - Get Affiliate commission plans
    pub async fn list_commission_plans(&self, params: Option<&serde_json::Value>) -> Result<serde_json::Value, crate::http_client::ApiError> {
        self.http.get("/api/affiliates/commissionplans", None).await
    }

}
